### PR TITLE
Adding verilator CI smoke test

### PR DESCRIPTION
**Pull request by howardtr**
- Copied mostly from https://github.com/chipsalliance/caliptra-sw/blob/main/.github/workflows/build-test-verilator.yml
  - Downgrade verilator from 5.006 to 4.228 since there are lint errors. We can fix this in a subsequent PR.
  - TODO: There's a step in the action which does a replacement of `Caliptra/` in the directory since there's a mismatch in how the path is used in the Makefile / caliptra_top_tb.vf, which causes a conflict.  This should be removed when that is resolved

Note: A later version of the riscv toolchain is used here since I had trouble building
riscv64-unknown-elf-gcc-8.2.0-2019.02 from source.

### CODE REVIEW

File: src/integration/config/caliptra_top_tb.vf
Comment (calebofearth) [inline src/integration/config/caliptra_top_tb.vf:1]: Since a similar change is introduced in #21, can we remove the changes here? To avoid conflicts 😊

**Review by calebofearth — APPROVED** [review]
If we can complete this PR without including caliptra_top_tb.vf, that would be ideal. Otherwise, approved.

File: .github/workflows/build-test-verilator.yml [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+on:
+  push:
+    branches: ["main", "dev-goog", "dev-msft"]
+  pull_request:
+
+  workflow_dispatch:
+
+jobs:
+  build_and_test:
+    name: Verilator Build and Test
```
Comment (korran) [inline .github/workflows/build-test-verilator.yml:14]: Call this Verilator Smoke Test

File: .github/workflows/build-test-verilator.yml [hunk 2]
```diff
@@ -0,0 +1,126 @@
+# docs: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions
+
+name: Verilator Hardware Model
```
Comment (korran) [inline .github/workflows/build-test-verilator.yml:3]: Call this Verilator

File: src/integration/config/caliptra_top_tb.vf
Comment (howardtr) [reply src/integration/config/caliptra_top_tb.vf:1]: yup, let me take care of that right now

**Review by korran — APPROVED** [review]

File: src/integration/config/caliptra_top_tb.vf
Comment (howardtr) [reply src/integration/config/caliptra_top_tb.vf:1]: So it looks like the Makefile and src/integration/config/caliptra_top_tb.vf weren't in sync

Makefile was looking at `$(WORKSPACE)/src/integration/test_suites` while caliptra_top_tb.vf was looking at `${WORKSPACE}/Caliptra/src/`

So setting WORKSPACE=some_root, was giving conflicts.

As a workaround, I added a sed step to replace /Caliptra/ in the GitHub Action that we can replace in your integration PR

File: .github/workflows/build-test-verilator.yml [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+      - name: Save verilator dir
+        uses: actions/cache/save@v3
+        if: steps.verilator_restore.outputs.cache-hit != 'true'
+        with:
+          path: /opt/verilator
+          key: verilator-${{ env.VERILATOR_VERSION }}-${{ env.SCCACHE_C_CUSTOM_CACHE_BUSTER }}
+
+      - name: Setup verilator path
+        run: |
+          echo /opt/verilator/bin >> $GITHUB_PATH
```
[the remark this answers is not in the document]
Comment (calebofearth) [reply .github/workflows/build-test-verilator.yml:92]: Nevermind - did a quick web search on GITHUB_PATH and learned something :)

**Review by calebofearth — APPROVED** [review]

**Review by korran — APPROVED** [review]

**Pull request merged**